src/NLog/Targets/FileTarget.cs

```diff
@@ -562,27 +570,35 @@ public int MaxArchiveFiles
         /// <docgen category='Archival Options' order='10' />
         public ArchiveNumberingMode ArchiveNumbering { get; set; }
 
-#if NET4_5
+        /// <summary>
+        /// Per instance file compressor to be used to compress log files during archiving.
+        /// Defaults to <see cref="DefaultCompressor"/>.
+        /// </summary>
+        public IFileCompressor Compressor { get; set; }
```

Comment:
That would have been my personal preference, but you proposed "Compressor" as the property name above ;)